### ISSUE TITLE
Moving documentation to PowerShell Docs

### ISSUE THREAD

**Issue by johlju** [PowerShell/DscResources#263]
During the community call I raised the question if we could move (bit by bit) the resource documentation to PowerShell Docs repo. I couldn't find the issue I raised this in so I writing a new issue. :)
This would then be instead of having the documentation on GitHub Wiki.

With resource documentation I mean all the documentation for the resource module, like descriptions what a resource does, it's parameters with their descriptions, examples and so on. Document it much like how a cmdlet is documented today in PowerShell Docs, something like [Get-Service](https://msdn.microsoft.com/en-us/powershell/reference/5.1/microsoft.powershell.management/get-service). Of course there probably need to be a different layout for documenting a resource.

I also see a demand for step-by-step guides (issue https://github.com/PowerShell/xSQLServer/issues/176) which for xSQLServer means we need to incorporate other resources like xFailoverCluster in that guide. We have a lot of examples over at xSQLServer which should be documented more in a blog style (issue https://github.com/PowerShell/xSQLServer/issues/462), which got me thinking again of PowerShell Docs.
Having these in PowerShell Docs we could use all knownledge of the PowerShell community to keep these up-to-date. 
It would be harder for the community to find these, and update these, in the module folder or in GitHub wiki when there is a new good looking doc site with "all other" PowerShell documentation on it. :)

So the first question is. Is it possible for us to start moving/creating documentation on PowerShell Docs?

And if that is a yes, then we need to figure out how to do that practically, or if it is even practically. 
Will it mean that we need to ask the contributor to send in a PR to the PowerShell Docs repo when there is a need for a change in the current documentation? Or can we automate that? 
Can we have control over deploy to the Docs site, so instead of the DocumentHelper documenting on GitHub Wiki it updates the Docs site? Or maybe it sends in a PR to the PowerShell Docs repo?

Either way PowerShell Docs repo will become the source from which all updates must be based upon.

I see there would be much potential have the documentation in the PowerShell Docs. To get out to a broader community, with even more knowledge.

**Comment by PlagueHO** [comment]
I agree with @johlju - this sounds like it could be a pretty good idea.

But I'd suggest we'd only want to do this with the repos that currently support Wiki/Autodocumentation (once [this PR](https://github.com/PowerShell/DscResource.Tests/pull/131) has been merged). We'd also want to completely automate the process of copying the Wiki content over when the module is published to the PSGallery - this should be reasonably easy to automate as part of the release process (but would require the PS Team @kwirkykat / @mbreakey3 to create the process I'd expect).

So tl;dr:
- Wiki repos only (I'm working on changing other repos to Wiki)
- Publish docs by automation only.

My 2c!

**Comment by zjalexander** [comment]
Sorry, I missed the community call due to the PowerShell summit and then I was out sick for a few days. 
I'm the doc owner for PowerShell and the right person to investigate and come back with answers. 

>So the first question is. Is it possible for us to start moving/creating documentation on PowerShell Docs?

Great question. We think about docs in two categories  - "reference" (basically what you would get via get-help and API docs, etc), and "conceptual" (your example of SQL server) 

Right now, our reference content only covers PowerShell core cmdlets that we publish and maintain. Having community module reference content would be a change, so let me talk to doc writers and see what we come up with. I agree we have a particularly fragmented story on the DSC side of things right now and some action is needed.

For the conceptual content, I think it would be great to have stuff like the step-by-step guide to setting up SQL on our docs page. We badly need more guides of this nature. cc @eslesar as he's investigating conceptual content right now - I'll also talk to him in person. 

I have some minor concerns about having community stuff listed on the Official Microsoft Documentation, but I'll dig around to see what other communities (Chef, Puppet, etc) do and what a good best practice would be in this instance.

**Comment by zjalexander** [comment]
Follow-up as we discussed this in the community call:
We are moving towards a model of PowerShell where each module should live on its own where possible and include its own documentation.

As a result, my reccomendation is to keep documentation in the same repo as the module and link to it from e.g. the gallery when necessary. This is similar to what Chef and Puppet do.

**Comment by johlju** [comment]
Closing this as per comment from @zjalexander. On top of his comment, we are also moving towards adding resource modules that are maintained inside the community's own repositories.
